### PR TITLE
Implement positioned read

### PR DESCRIPTION
**Pull request by calvinjia**
This PR also uses the newer API to do partial page reads correctly.

### CODE REVIEW

File: core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheFileInStream.java [hunk 1]
```diff
@@ -216,12 +249,14 @@ private FileInStream getExternalFileInStream(long pos) throws IOException {
    * Reads a page from external storage which contains the position specified. Note that this makes
    * a copy of the page.
    *
+   * This method is synchronized to ensure thread safety for positioned reads.
```
Comment (apc999) [inline core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheFileInStream.java]: not clear to me what are we guarding. in addition, do we only provide thread-safety for positioned reads not normal reads?

Comment (calvinjia) [reply core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheFileInStream.java]: Yeah, only for positioned reads (PositionedReadable). The read API (InputStream) is not thread safe.

Comment (calvinjia) [reply core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheFileInStream.java]: I explained a bit more in the comment.

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**